letsencrypt-auto-source/letsencrypt-auto.template

```diff
@@ -330,13 +330,27 @@ elif [ -f /etc/redhat-release ]; then
   prev_le_python="$LE_PYTHON"
   unset LE_PYTHON
   DeterminePythonVersion "NOCRASH"
+
   # Starting to Fedora 29, python2 is on a deprecation path. Let's move to python3 then.
+  # RHEL 8 also uses python3 by default.
   RPM_DIST_NAME=`(. /etc/os-release 2> /dev/null && echo $ID) || echo "unknown"`
-  RPM_DIST_VERSION=0
-  if [ "$RPM_DIST_NAME" = "fedora" ]; then
-    RPM_DIST_VERSION=`(. /etc/os-release 2> /dev/null && echo $VERSION_ID) || echo "0"`
+
+  # CentOS 6 does not have an /etc/os-release file.  On these systems RPM_DIST_VERSION will be set to 0.
+  if [ -f /etc/os-release ]; then
+    RPM_DIST_VERSION=`(. /etc/os-release 2> /dev/null && echo $VERSION_ID) | cut -d '.' -f1`
```

Comment:
My understanding based on https://www.freedesktop.org/software/systemd/man/os-release.html is that there is no requirement that `VERSION_ID` exists in `/etc/os-release` in which case I think `RPM_DIST_VERSION` still is not set to an integer that can be compared below. With this in mind, I think it's still possible for us to have the [failures we had before on CentOS](https://github.com/certbot/certbot/pull/7240#discussion_r310284518).

Is there a more robust way to write this? What do you think @schoen?

Reply:
Heck, it looks like we technically can't even count on `VERSION_ID` containing an integer according to the freedesktop link above which says:

> A lower-case string (mostly numeric, no spaces or other characters outside of 0–9, a–z, ".", "_" and "-") identifying the operating system version, excluding any OS name information or release code name, and suitable for processing by scripts or usage in generated filenames. This field is optional. Example: "VERSION_ID=17" or "VERSION_ID=11.04".

Reply:
@schoen and I brainstormed about this out of band and here's what we came up with:
```
  # Set RPM_DIST_VERSION to VERSION_ID from /etc/os-release after splitting on
  # '.' characters (e.g. "8.0" becomes "8"). If the command exits with an
  # error, RPM_DIST_VERSION is set to unknown.
  RPM_DIST_VERSION=$( (. /etc/os-release 2> /dev/null && echo "$VERSION_ID") | cut -d '.' -f1 || echo "unknown")
  # If RPM_DIST_VERSION is an empty string or it contains any nonnumeric
  # characters, the value is unexpected so we set RPM_DIST_VERSION to 0.
  if [ -z "$RPM_DIST_VERSION" ] || [ -n "$(echo "$RPM_DIST_VERSION" | tr -d '[0-9]')" ]; then
     RPM_DIST_VERSION=0
  fi
```

Reply:
That looks fine to me.  I'll update the PR tomorrow.  On a side note, this stuff would be a lot easier using facter.  :)

Reply:
Thanks again @blackknight36.

What's facter?

Reply:
@bmw facter is an application that provides node facts such as the operating system, OS release, and many other things.  It's used by Puppet but you can also run it standalone.

Reply:
I've updated the script with the changes suggested.

Reply:
Thank you.

I updated my PR which includes your latest changes at https://github.com/certbot/certbot/pull/7302.